[Bug]: external deployment is failing with timeout for pvc to bound

### Description
job: https://url.corp.redhat.com/b0c703a


```
2026-04-20 15:42:39      @retry((TemporaryPodsDuringDeployment, AssertionError), tries=3, delay=60, backoff=1)
2026-04-20 15:42:39      def validate_cluster_on_pvc():
2026-04-20 15:42:39          """
2026-04-20 15:42:39          Validate creation of PVCs for MON and OSD pods.
2026-04-20 15:42:39          Also validate that those PVCs are attached to the OCS pods
2026-04-20 15:42:39      
2026-04-20 15:42:39          Raises:
2026-04-20 15:42:39               AssertionError: If PVC is not mounted on one or more OCS pods or some of the PVCs are not bound
2026-04-20 15:42:39               TemporaryPodsDuringDeployment: If canary or temporary pods detected (triggers retry)
2026-04-20 15:42:39      
2026-04-20 15:42:39          """
2026-04-20 15:42:39          # Get the PVCs for selected label (MON/OSD)
2026-04-20 15:42:39          ns = config.ENV_DATA["cluster_namespace"]
2026-04-20 15:42:39      
2026-04-20 15:42:39          try:
2026-04-20 15:42:39              pvc_names = _collect_bound_ocs_pvcs(namespace=ns)
2026-04-20 15:42:39          except TimeoutExpiredError as exc:
2026-04-20 15:42:39  >           raise AssertionError("Timed out waiting for OCS PVCs to become Bound") from exc
2026-04-20 15:42:39  E           AssertionError: Timed out waiting for OCS PVCs to become Bound
2026-04-20 15:42:39  
2026-04-20 15:42:39  ocs_ci/ocs/cluster.py:1324: AssertionError
```

--- a/ocs_ci/ocs/cluster.py
+++ b/ocs_ci/ocs/cluster.py
@@ -1277,13 +1277,19 @@ def _collect_bound_ocs_pvcs(namespace, timeout=300, sleep=10):
 
     def _get_matching_bound_pvcs():
         ocs_pvc_obj = get_all_pvc_objs(namespace=namespace)
-        matching_pvcs = [
-            pvc_obj
-            for pvc_obj in ocs_pvc_obj
-            if pvc_obj.name.startswith(constants.DEFAULT_DEVICESET_PVC_NAME)
-            or pvc_obj.name.startswith(constants.DEFAULT_MON_PVC_NAME)
-            or any(pvc_obj.name.startswith(name) for name in storage_device_set_names)
-        ]
+        matching_pvcs = []
+        for pvc_obj in ocs_pvc_obj:
+            if pvc_obj.name.startswith(constants.DEFAULT_DEVICESET_PVC_NAME):
+                matching_pvcs.append(pvc_obj)
+            elif pvc_obj.name.startswith(constants.DEFAULT_MON_PVC_NAME):
+                matching_pvcs.append(pvc_obj)
+            elif pvc_obj.name.startswith(constants.NB_DB_CNPG_CLUSTER_NAME):
+                matching_pvcs.append(pvc_obj)
+            else:
+                for name in storage_device_set_names:
+                    if pvc_obj.name.startswith(name):
+                        matching_pvcs.append(pvc_obj)
+                        break
         not_bound = [
             pvc_obj.name
             for pvc_obj in matching_pvcs

